src/niri.rs

```diff
@@ -3821,6 +3944,18 @@ impl Niri {
         }
     }
 
+    pub fn remove_screencopy_output(&mut self, output: &Output) {
+        for queue in self.screencopy_state.queues_mut() {
+            if queue
+                .split()
+                .1
+                .is_some_and(|screencopy| screencopy.output() == output)
+            {
+                queue.pop();
+            }
```

Comment:
I guess this needs to check *all* frames in the queue. If any queued frame matches the output, pop all frames starting from that one to the end of the queue.